### PR TITLE
Changes over AR Querying guide making use of Bookstore example

### PR DESCRIPTION
**Pull request by vipulnsward**
Closes #35008

@weedySeaDragon has added so many improvements, going over took some time.

What:
- Changes over AR Querying guide making use of Bookstore example and cleaning up multiple re-declarations of models and cleaning up multiple re-declarations of models
- Uses bookstore models for all examples
- Add diagram changes
- Add Output changes
- Expansion of Enum documentation
- Lots of Grammar fixes
- [x] Needs a changelog entry

Inspiration:
- See https://github.com/rails/rails/pull/35008 description for inspiration

Changes from original PR:
- Removed some examples get first version merged(like has_reviews). We can add more if needed later
- Culled down some examples from Enum docs 
- Incorporated review changes from @gmcgibbon 
- NOTE: ANDed should just be that(https://english.stackexchange.com/questions/79659/the-verb-for-carrying-out-a-bitwise-or-and-operation)

This will certainly have follow up edits, trying to get the first version in so subsequent cleanups are smaller.

Live Version of this guide for easy viewing: 
https://ar-querying-guide.netlify.app/active_record_querying.html

Co-authored-by: Ashley Engelund (@weedySeaDragon)

### CODE REVIEW

**Comment by weedySeaDragon** [comment]
@vipulnsward Thanks for the work!  I know it was a big, honkin' PR.  But I hope that ultimately it help folks.

**Comment by memoht** [comment]
@weedySeaDragon I'm rooting for you.

**Comment by fxn** [comment]
Hey! I have read a bit, will continue.

Let me share an early remark. The `Book` model defines this association:

```ruby
belongs_to :author, dependent: :destroy
```

I believe that is not natural. It is rare that you want to delete an author when you delete their books. Even less in an introductory guide. That option normally goes in the other direction: when you delete an author, you delete all their books, and their pictures, hearts given by users, ..., all related models.

Even more, unless you define FK constraints with cascade delete, you rarely want to declare a `has_many` without `:dependent`, because by default AR does nothing and you'd leave orphan records.

This guide should be a model for these kind of best practices.

**Comment by weedySeaDragon** [comment]
I'm not sure what the next steps are; not sure who needs to review and approve this.  Perhaps it's you, @vipulnsward ?  
(I'm commenting so this doesn't get closed).

**Comment by vipulnsward** [comment]
Thanks @rafaelfranca ! I will work over cleaning this more soon.

File: guides/source/active_record_querying.md [hunk 1]
```diff
@@ -1608,46 +1681,44 @@ a single object (see [Retrieving a Single Object Section](#retrieving-a-single-o
...
 
 The above should generate:
 
 ```sql
-SELECT people.id, people.name, companies.name
-FROM people
-INNER JOIN companies
-  ON companies.person_id = people.id
-WHERE people.name = 'John'
-LIMIT 1
```
Comment (ghiculescu) [inline guides/source/active_record_querying.md:1650]: not sure if this line was meant to be removed, but assuming no so adding it back in https://github.com/rails/rails/pull/40349

**Pull request merged**